lib/spack/spack/compilers/msvc.py

```diff
@@ -103,11 +103,21 @@ def short_msvc_version(self):
         """
         This is the shorthand VCToolset version of form
         MSVC<short-ver> *NOT* the full version, for that see
-        Msvc.msvc_version
+        Msvc.msvc_version or MSVC.platform_toolset_ver for the
+        raw platform toolset version
         """
-        ver = self.msvc_version[:2].joined.string[:3]
+        ver = self.platform_toolset_ver
         return "MSVC" + ver
 
+    @property
+    def platform_toolset_ver(self):
+        """
+        This is the platform toolset version of current MSVC compiler
+        of form v<toolset-ver>, i.e. v142. This is different from the VC
+        toolset version as established by `short_msvc_version`
+        """
+        return self.msvc_version[:2].joined.string[:3]
```

Comment:
If this should be returning for example "v142", then I'll note that `[:3]` takes the first 3 characters. If it in fact should be "142" then this looks right (although I think the docstring should be clarified in that case).

Reply:
It should only be 142, and packages are left to prepend the 'v' character if necessary. I'll update the docstring.